graphql_compiler/compiler/match_lowering/match_between_lowering.py

```diff
@@ -0,0 +1,121 @@
+from collections import deque
```

Comment:
I would prefer to not include `match` in the name of this file -- the name of the directory (package) that it's in already communicates that adequately. I think `between_lowering.py` is probably fine.